controllers/resources/namespaced.go

```diff
@@ -21,63 +23,75 @@ import (
 
 type Namespaced struct {
 	client    client.Client
-	finalizer Processor
+	processor Processor
 }
 
 func (r *Namespaced) SetupWithManager(mgr ctrl.Manager) error {
-	unstructuredCachingClient, err := client.NewDelegatingClient(
-		client.NewDelegatingClientInput{
-			Client:            mgr.GetClient(),
-			CacheReader:       mgr.GetCache(),
-			CacheUnstructured: true,
-		},
-	)
-	if err != nil {
-		return err
-	}
-
-	r.client = mgr.GetClient()
-	r.finalizer = Processor{
-		client:             r.client,
-		unstructuredClient: unstructuredCachingClient,
+	r.processor = Processor{
+		client: mgr.GetClient(),
 	}
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&capsulev1beta2.TenantResource{}).
 		Complete(r)
 }
 
+//nolint:dupl
 func (r *Namespaced) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log := ctrllog.FromContext(ctx)
 
 	log.Info("start processing")
 	// Retrieving the TenantResource
-	tntResource := capsulev1beta2.TenantResource{}
-	if err := r.client.Get(ctx, request.NamespacedName, &tntResource); err != nil {
-		if apierr.IsNotFound(err) {
+	tntResource := &capsulev1beta2.TenantResource{}
+	if err := r.client.Get(ctx, request.NamespacedName, tntResource); err != nil {
+		if apierrors.IsNotFound(err) {
 			log.Info("Request object not found, could have been deleted after reconcile request")
 
 			return reconcile.Result{}, nil
 		}
 
-		log.Error(err, "cannot retrieve capsulev1beta2.TenantResource")
-
 		return reconcile.Result{}, err
 	}
+
+	patchHelper, err := patch.NewHelper(tntResource, r.client)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to init patch helper")
+	}
+
+	defer func() {
+		if e := patchHelper.Patch(context.TODO(), tntResource); e != nil {
```

Comment:
Same here, let's use the `context` from the reconcile request.

Reply:
done